promql/functions.go

```diff
@@ -184,6 +184,109 @@ func funcIrate(ev *evaluator, args Expressions) model.Value {
 	return resultVector
 }
 
+// Calculates holt-winters smoothed value at the given index.
+// The smoothed value is created by creating two values (x,y) where...
+// x = the current raw data point, weighted by the smoothing factor.
+// y = the last smoothed point + the last trend value weighted by the inverse of the smoothing factor.
+// s = computed smoothed values, b = computed trend factors, d = raw
+func doubleSVal(i int, sf, tf float64, s, b, d []float64) float64 {
+
+	x := sf * d[i]
+	y := (1 - sf) * (s[i-1] + doubleBVal(i-1, sf, tf, s, b, d))
+
+	// cache value
+	s[i] = x + y
```

Comment:
This is duplicating what `funcHoltWinters` is doing.  I'd suggest moving that for loop into here, and just returning the result.